metastore/src/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java

```diff
@@ -569,7 +569,7 @@ public Partition alterPartition(final RawStore msdb, Warehouse wh, final String
           LOG.error("Revert the data move in renaming a partition.");
           try {
             if (destFs.exists(destPath)) {
-              wh.renameDir(destPath, srcPath);
+              wh.renameDir(destPath, srcPath, false);
```

Comment:
As discussed, ignoring this comment as we use the flag to decide if we need to recycle or not.